### ISSUE TITLE
[Feature]: Auto width for widgets

### ISSUE THREAD

**Issue by pranavkanade** [appsmithorg/appsmith#21729]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Similar to Auto height, a user has asked for an auto width feature.

### Why should this be worked on?

In their use case, they have a two-column layout and when one of the widgets is invisible, they see empty space. Instead, they want the adjacent widget to expand sideways to take-up the space.

**Comment by tlawrence-TAG** [comment]
Yes, agree with this ask. If not auto, add the ability for widgets to be collapsed. Also, auto-height does seem to take up some CPU cycles... ie.. it's much slower to render than fixed. I've stopped using it because of this.
